### PR TITLE
test(store): add dtslint

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by markwhitfeld** [comment]
Will have a look tomorrow. Will probably need to add the dtslint task to the circle ci config as well.

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
   "scripts": {
     "// - Binaries": "Run against node_modules/.bin",
     "ng": "ng",
+    "types": "types",
```
Comment (markwhitfeld) [inline package.json]: What is this script used for?

File: package.json [hunk 2]
```diff
@@ -117,6 +119,7 @@
     "codelyzer": "^4.5.0",
     "core-js": "^2.6.0",
     "cross-env": "^5.2.0",
+    "dtslint": "^0.6.1",
```
Comment (markwhitfeld) [inline package.json]: We need to run this from Circle CI too.

File: package.json (hunk 1 above)
Comment (splincode) [reply package.json]: removed

File: package.json (hunk 2 above)
Comment (splincode) [reply package.json]: done

File: .travis.yml [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
 
 cache: yarn
 
 install:
   - yarn install
 
 script:
-  # CHECK STYLE
-  - yarn run lint
-
```
Comment (markwhitfeld) [inline .travis.yml]: What was the reason for removing this from the Travis build?

Comment (splincode) [reply .travis.yml]: I don’t really understand, but for some reason there is a conflict on travis and errors on the lint fall

Comment (splincode) [reply .travis.yml]: image

But Circle CI works

File: yarn.lock [hunk 4]
```diff
@@ -8514,6 +8530,25 @@ tslib@^1.8.0, tslib@^1.8.1, tslib@^1.9.0, tslib@^1.9.3:
   resolved "https://registry.npmjs.org/tslib/-/tslib-1.9.3.tgz#d7e4dd79245d85428c4d7e4822a79917954ca286"
   integrity sha512-4krF8scpejhaOgqzBEcGM7yDIEfi0/8+8zDRZhNZZ2kjmHJ4hv3zCbQWxoJGz1iw5U0Jl0nma13xzHXcncMavQ==
 
+tslint@5.14.0:
```
Comment (markwhitfeld) [inline yarn.lock]: @splincode I see that we are running 2 versions of tslint here. Maybe that is causing the travic CI issues?
We could pin it to 5.14, which is required by dtslint.
Please also add the lint task back to the travis build... I would like to try to get this working...

Comment (splincode) [reply yarn.lock]: Yes!

Comment (markwhitfeld) [reply yarn.lock]: Do you think that is the issue?

Comment (splincode) [reply yarn.lock]: I need investigation

File: .travis.yml [hunk 5]
```diff
@@ -1,17 +1,14 @@
 language: node_js
 sudo: false
 node_js:
-  - "11.4.0"
+  - '11.4.0'
 
 cache: yarn
 
 install:
   - yarn install
```
Comment (markwhitfeld) [inline .travis.yml]: The source of the discrepancy between circle CI and Travis is probably the fact that travis is not using the lockfile like Circle CI is doing. Circle CI has this command:
`yarn install --frozen-lockfile --non-interactive`
We should use this here to make both consistent.

File: .travis.yml (hunk 3 above)
Comment (markwhitfeld) [reply .travis.yml]: We still need to add this back so that we are getting the full picture of this failure.

File: .travis.yml (hunk 5 above)
Comment (splincode) [reply .travis.yml]: done

File: yarn.lock (hunk 4 above)
Comment (splincode) [reply yarn.lock]: done

File: .travis.yml (hunk 3 above)
Comment (splincode) [reply .travis.yml]: done

**Pull request merged**